framework/13-inch/7040-amd/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, pkgs, ... }: {
+  imports = [
+    ../../../common/cpu/amd
+    ../../../common/cpu/amd/pstate.nix
+    ../../../common/gpu/amd
+    ../../../common/pc/laptop
+    ../../../common/pc/laptop/ssd
+  ];
+
+  # Newer kernel is better for amdgpu driver updates
+  # Requires at least 5.16 for working wi-fi and bluetooth (RZ616, kmod mt7922):
+  # https://wireless.wiki.kernel.org/en/users/drivers/mediatek
+  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "6.1") (lib.mkDefault pkgs.linuxPackages_latest);
+
+  # AMD has better battery life with PPD over TLP:
+  # https://community.frame.work/t/responded-amd-7040-sleep-states/38101/13
+  services.power-profiles-daemon.enable = lib.mkDefault true;
+
+  # Fix TRRS headphones missing a mic
+  # https://community.frame.work/t/headset-microphone-on-linux/12387/3
+  boot.extraModprobeConfig = ''
+    options snd-hda-intel model=dell-headset-multi
```

Comment:
JFYI; this sounds like it should be a kernel patch.  After someone writes up a patch and it lands somewhere you can later remove it.

Reply:
Yeah, probably - it's also been in the config since https://github.com/NixOS/nixos-hardware/commit/34485f1807befc9b104e59e0716e7a5cd0967fcb:
https://github.com/NixOS/nixos-hardware/blob/master/framework/default.nix#L20

(And might be an indication that there would be a benefit to refactoring these configs to share some kind of common base?)

Reply:
Added a note about it being a temporary configuration - as for common modules for framework, I plan to make a separate PR for that